src/sqlancer/common/ast/newast/NewBinaryOperatorNode.java

```diff
@@ -5,10 +5,10 @@
 public class NewBinaryOperatorNode<T> implements Node<T> {
 
     protected final Operator op;
-    protected final Node<T> left;
-    protected final Node<T> right;
+    protected final T left;
```

Comment:
I'm wondering about the semantics of this change (as well as the current implementation). Conceptually, I think that an AST consists of nodes that in turn have nodes as children. Based on this, should the type of this node not be the same as its children? The same would apply also to the other nodes.